### ISSUE TITLE
error when starting BlockUIService in ngOnInit

### ISSUE THREAD

**Issue by JulienOuvrard** [kuuurt13/ng-block-ui#120]
I'm currently trying to upgrade my application to Angular 9 and when starting the ng-block-ui service in an ngOnInit hook i get this error in my console: 

```
ng-block-ui: TypeError: Cannot convert undefined or null to object
    at Function.from (<anonymous>)
    at BlockUIDirective.push../node_modules/ng-block-ui/__ivy_ngcc__/fesm5/ng-block-ui.js.BlockUIDirective.isComponentInTemplate (ng-block-ui.js:836)
    at BlockUIDirective.push../node_modules/ng-block-ui/__ivy_ngcc__/fesm5/ng-block-ui.js.BlockUIDirective.ngOnInit (ng-block-ui.js:802)
    at callHook (core.js:2614)
    at callHooks (core.js:2584)
    at executeInitAndCheckHooks (core.js:2537)
    at selectIndexInternal (core.js:6141)
    at Module.ɵɵadvance (core.js:6114)
    at RootComponent_Template (root.component.html:54)
    at executeTemplate (core.js:7495)
```

BlockUIModule.forRoot() declared at top level module.
The BlockUIDirective is used in my RootComponent.

Config:

- Angular: 9.0.1
- ng-block-ui: 3.0.0-beta.4

My suggestion is that this code:
```
private isComponentInTemplate(element: any): boolean {
    let { children } = element || [];
    children = Array.from(children).reverse();
    return children.some((el: any) => el.localName === 'block-ui');
  }
```

can be changed to 

```
private isComponentInTemplate(element: any): boolean {
    let { children = [] } = element;
    children = Array.from(children).reverse();
    return children.some((el: any) => el.localName === 'block-ui');
  }
```

I did not wanted to do a pull request for one line change.

**Comment by JulienOuvrard** [comment]
FYI:
When compiling with target es5 (my application have to be IE11 friendly) the portion of code i suggest to change is compiled to :
```
BlockUIDirective.prototype.isComponentInTemplate = /**
     * @param {?} element
     * @return {?}
     */
function (element) {
        var children = (element || []).children;
        children = Array.from(children).reverse();
        return children.some(function (el) { return el.localName === 'block-ui'; });
};
```
and with my suggestion:
```
BlockUIDirective.prototype.isComponentInTemplate = /**
     * @param {?} element
     * @return {?}
     */
function (element) {
        var children = element.children || [];
        children = Array.from(children).reverse();
        return children.some(function (el) { return el.localName === 'block-ui'; });
};
```

children is always valued.

**Comment by kuuurt13** [comment]
@JulienOuvrard thanks for the details. I actually had a fix for this in my local working branch but didn't get it rolled out in `beta.4`. I have since added a fix and published `3.0.0-beta.5` please install and let me know if you have any issues (`npm i ng-block-ui@next`)

**Comment by JulienOuvrard** [comment]
@kuuurt13 with the 3.0.0-beta.5 the error is gone.

I was getting another error:
```
ng-block-ui.js:823 ng-block-ui: TypeError: Cannot read property 'add' of undefined
    at DefaultDomRenderer2.push../node_modules/@angular/platform-browser/__ivy_ngcc__/fesm5/platform-browser.js.DefaultDomRenderer2.addClass (platform-browser.js:710)
    at BaseAnimationRenderer.push../node_modules/@angular/platform-browser/__ivy_ngcc__/fesm5/animations.js.BaseAnimationRenderer.addClass (animations.js:258)
    at BlockUIDirective.push../node_modules/ng-block-ui/__ivy_ngcc__/fesm5/ng-block-ui.js.BlockUIDirective.ngOnInit (ng-block-ui.js:803)
    at callHook (core.js:2614)
    at callHooks (core.js:2584)
    at executeInitAndCheckHooks (core.js:2537)
    at selectIndexInternal (core.js:6141)
    at Module.ɵɵadvance (core.js:6114)
    at RootComponent_Template (root.component.html:58)
    at executeTemplate (core.js:7495)
```
with this markup directly inside my root.component.html:
```
<div *blockUI="'BlockRoot'" id="blockRootDiv">
    <!-- some components -->
</div>
```

I was able to fixed it by wrapping a `div` arround the blockUI directive
```
<div>
    <div *blockUI="'BlockRoot'" id="blockRootDiv">
        <!-- some components -->
    </div>
</div>
```

**Comment by kuuurt13** [comment]
@JulienOuvrard good to hear that it fixed your initial error. 

As for the issue with the directive in your root component, I have tried to replicate the error but can't seem to be able to. I have created a new Angular 9 app with the cli and I have added the directive to the outermost `div` in my root component and everything seems to run fine (see below). I also tested in `IE 11` and didn't run into any issues. Is there any other details that you could give me or do you have a repo I can look at?

### app.component.ts
```ts
import { Component, OnInit } from '@angular/core';
import { BlockUIService } from 'ng-block-ui';

@Component({
  selector: 'app-root',
  templateUrl: './app.component.html',
  styleUrls: ['./app.component.css']
})
export class AppComponent implements OnInit  {
  name = 'Test';

  constructor(private blockUIService: BlockUIService) { }

  ngOnInit() {
    this.blockUIService.start('test')
  }
}

```
### app.component.html
```html
<div *blockUI="'test'" id="tester">
  <hello name="{{ name }}"></hello>
</div>
```

### Angular CLI
```
Angular CLI: 9.0.2
Node: 12.0.0
OS: darwin x64

Angular: 9.0.1
... animations, common, compiler, compiler-cli, core, forms
... language-service, platform-browser, platform-browser-dynamic
... router
Ivy Workspace: Yes

Package                           Version
-----------------------------------------------------------
@angular-devkit/architect         0.900.2
@angular-devkit/build-angular     0.900.2
@angular-devkit/build-optimizer   0.900.2
@angular-devkit/build-webpack     0.900.2
@angular-devkit/core              9.0.2
@angular-devkit/schematics        9.0.2
@angular/cli                      9.0.2
@ngtools/webpack                  9.0.2
@schematics/angular               9.0.2
@schematics/update                0.900.2
rxjs                              6.5.4
typescript                        3.7.5
webpack                           4.41.2
```

**Comment by westor21** [comment]
I'm having the same issue:

Here is my code with the wrqapping div, which is working:

```
<ng-container *ngIf="shouldRenderApplication">
  <ng-template #uzBlockTemplate>
    <div class="main-blocker"></div>
  </ng-template>
  <div *ngIf="routingInTransition" class="main-blocker"></div>
  <!--following div is a workaround or block-ui error-->
  <div>
    <div *blockUI="'main-blocker'; template: uzBlockTemplate">
      <div>
        <router-outlet></router-outlet>
      </div>
    </div>
  </div>
</ng-container>
<ng-container *ngIf="!shouldRenderApplication">
<div>
    Loading Application...
</div>
</ng-container>

```
controller:

```
  constructor(private permissionService: PermissionService,
              private titleService: Title,
              private router: Router
  ) {
    this.routeSubscription = router.events
      .subscribe((event: RouterEvent) => this.navigationInterceptor(event));
  }

  navigationInterceptor(event: RouterEvent): void {
    if (event instanceof NavigationStart) {
      this.routingInTransition = true;
    } else if (event instanceof NavigationEnd || event instanceof NavigationCancel || event instanceof NavigationError) {
      setTimeout(() => this.routingInTransition = false);
    }
  }

```
Without the wrapping div I get this:
```
ng-block-ui.js:789 ng-block-ui: TypeError: Cannot convert undefined or null to object
    at Function.from (<anonymous>)
    at BlockUIDirective.isComponentInTemplate (ng-block-ui.js:798)

```
children is undefined here:

![image](https://user-images.githubusercontent.com/1253973/75096049-a2edb280-559b-11ea-9c92-0a1c51919b14.png)

**Comment by kuuurt13** [comment]
@westor21 thanks for all the info! I was able to reproduce your issue and I believe I have a fix. Please install the latest beta (`3.0.0-beta.6`) and let me know if it fixed your issues.

**Comment by JulienOuvrard** [comment]
@kuuurt13 Unfortunately, I was not able to reproduce either with the minimal repo, and I cannot make you see the actual code of my application because it contains some sensitive data of my client.

All I can say is that, as @westor21 has shown, i have a dynamic div juste before my ngBlockUi directive.

Waiting for beta.6 actively to test it 😃

**Comment by kuuurt13** [comment]
@JulienOuvrard not sure if you saw my comment above, `3.0.0-beta.6` has been released (`npm i ng-block-ui@next`)

**Comment by JulienOuvrard** [comment]
@kuuurt13 I realized my mistake just after replying, i will test the beta.6 asap 😉

**Comment by kuuurt13** [comment]
Fixed by #113
